### PR TITLE
Feature/new pipeline.code only

### PR DESCRIPTION
**Pull request by monkey-jeff**
Just the code changes for #257 

Review here but merge there.  Due to api refactor the change list was far too much.

### CODE REVIEW

File: src/lib/templateHelpers.test.ts [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+  });
+
+  it("Will not prefix object with a namespace", () => {
+    expect(addNamespace("object", "types"))
+      .to.be.a("string")
+      .and.equal("object");
+  });
+
+  it("Will not prefix Object with a namespace", () => {
+    expect(addNamespace("object", "types"))
```
Comment (kalagarraj) [inline src/lib/templateHelpers.test.ts:35]: are you trying to test "Object"?

File: src/lib/templateHelpers.test.ts [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    expect(addNamespace("foo", "types"))
+      .to.be.a("string")
+      .and.equal("types.foo");
+  });
+
+  it("Will not prefix void with a namespace", () => {
+    expect(addNamespace("void", "types")).to.be.a("string").and.equal("void");
+  });
+
+  it("Will not prefix object with a namespace", () => {
```
Comment (kalagarraj) [inline src/lib/templateHelpers.test.ts:28]: I see duplicate tests with different description, Are you trying to test for case insensitive string "object"?

File: src/lib/templateHelpers.test.ts [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+      .and.equal("object");
+  });
+
+  it("Will not prefix Object with a namespace", () => {
+    expect(addNamespace("object", "types"))
+      .to.be.a("string")
+      .and.equal("object");
+  });
+
+  it("Throws an error when called with undfined content", () => {
```
Comment (kalagarraj) [inline src/lib/templateHelpers.test.ts:40]: undfined typo

File: src/lib/templateHelpers.test.ts [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    expect(() => addNamespace(null, "types")).to.throw("Invalid content");
+  });
+
+  it("Throws an error when called with null content", () => {
+    expect(() => addNamespace(undefined, "types")).to.throw("Invalid content");
+  });
+
+  it("Throws an error when the type is not a valid string", () => {
```
Comment (kalagarraj) [inline src/lib/templateHelpers.test.ts:48]: add few more tests for empty string, string with only spaces

File: src/lib/templateHelpers.test.ts [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+});
+
+describe("When adding namespaces to elements in a complex content (an array)", () => {
+  it("Prefixes each element with a namespace", () => {
+    expect(addNamespace("Array<Foo | Baa>", "types"))
+      .to.be.a("string")
+      .and.equal("Array<types.Foo | types.Baa>");
+  });
+
+  it("Correctly parses three elements in an array", () => {
```
Comment (kalagarraj) [inline src/lib/templateHelpers.test.ts:60]: Add a test for zero and one element as well with a namespace

File: src/lib/templateHelpers.ts [hunk 6]
```diff
@@ -4,35 +4,66 @@
...
- * @returns true if the parameter is a common parameter
- */
-export const isCommonQueryParameter = (property: string): boolean =>
-  property
-    ? commonParameterPositions.queryParameters.includes(property.toString())
-    : false;
+  // if the content is an array, extract all of the elements
+  const matched = content.match(/^Array<(.*?)>$/);
+  const arrayType = !!matched;
+  const types = matched?.[1] || content;
```
Comment (kalagarraj) [inline src/lib/templateHelpers.ts:34]: potential index out of bounds error.. do u want to check if the length is greater than 1 before getting the index 1

File: src/lib/templateHelpers.ts [hunk 7]
```diff
@@ -103,5 +98,26 @@ export const formatForTsDoc = (str: string): string => {
...
+/**
+ * Checks if a path parameter is one of the set that are configurable at the client level
+ *
+ * @param property - The string name of the parameter to check
+ *
+ * @returns true if the parameter is a common parameter
+ */
+export const isCommonPathParameter = (property: string): boolean =>
+  property
+    ? commonParameterPositions.pathParameters.includes(property.toString())
```
Comment (kalagarraj) [inline src/lib/templateHelpers.ts:110]: Why do we need to call toString() on string?

File: src/lib/templateHelpers.test.ts (hunk 1 above)
Comment (monkey-jeff) [reply src/lib/templateHelpers.test.ts:35]: This test was moved wholesale.  Originally written by @dbarbato-sf

File: src/lib/templateHelpers.test.ts (hunk 4 above)
Comment (monkey-jeff) [reply src/lib/templateHelpers.test.ts:48]: This was moved wholesale. I would prefer not to much else beyond the refactor to keep it focused.

File: src/lib/templateHelpers.test.ts (hunk 5 above)
Comment (monkey-jeff) [reply src/lib/templateHelpers.test.ts:60]: Same comment

File: src/lib/templateHelpers.ts (hunk 6 above)
Comment (monkey-jeff) [reply src/lib/templateHelpers.ts:34]: This was also copied over.  This might be an issue but we should confirm.  Also written by @dbarbato-sf

File: src/lib/templateHelpers.ts (hunk 7 above)
Comment (monkey-jeff) [reply src/lib/templateHelpers.ts:110]: Don't know, didn't change this.

File: src/lib/templateHelpers.ts (hunk 6 above)
Comment (wjhsf) [reply src/lib/templateHelpers.ts:34]: JavaScript doesn't throw if you access an unset index, it just returns `undefined`. This is safe.
